### DISCUSSION TITLE
Asahi Linux OpenGL work and UTM?

### DISCUSSION THREAD

**Discussion by ajmas** [utmapp/UTM#4796] [General]
There is work being done on the Asahi Linux side to provide [OpenGL support](https://asahilinux.org/2022/12/gpu-drivers-now-in-asahi-linux/) on ARM based Macs.

I am curious whether any of this work can be leveraged by UTM, to improve on the graphic acceleration support?

**Comment by osy** [comment]
There’s no GPU passthrough support on macOS and QEMU’s virtual hardware has no relation to actual M1 hardware.

**Comment by ajmas** [comment]
Is this something that is being looked at on the QEMU side? 

I do see some old articles on QEMU and passthrough, but they seem to be more based on the Intel implementation.

**Reply by osy** [reply]
It has nothing to do with QEMU. "There’s no GPU passthrough support on macOS"

**Comment by MichelDiz** [comment]
Hey everyone, let me share my opinion on this.

I firmly believe it would be **99% possible** to install **Fedora Asahi Remix** (aka Asahi Linux) in a **UTM VM**. Why? Because you’ve already managed to run native macOS with graphics acceleration in a VM! The same approach could be applied here: using a similar bootloader setup (like OpenCore) to allow booting from a different disk. Of course, if there isn’t a more straightforward method.

So I thought about running the script **inside a virtualized macOS - in UTM**. That way, it could create the necessary partition inside the VM, and maybe I could dual-boot within the VM itself.

```bash
curl https://alx.sh | sh
```

but it didn’t work, because the script checks the OS version and hardware, and prevents any attempt to proceed. That could be another hurdle. (I also couldn’t find an official ISO — they only provide the script.)

But I realized that this setup lacks the macOS bootloader functionality (like recovery mode), so switching the boot inside the VM wouldn’t be possible this way(that's why I mentioned OpenCore). Still, it would have been an interesting approach. instead of dual-booting on the physical machine, running it directly inside a VM.

After analyzing all this, I do believe it’s definitely possible to **virtualize Asahi Linux** in this way. Cuz you can do it with macOS it self. And that my friends, is hardware passthrough.
